src/modm/platform/comp/stm32/comp.hpp.in

```diff
@@ -19,696 +17,435 @@
  * @defgroup	{{partname}}_comp Comparator
  */
 
-#if defined(STM32F3XX)
-namespace modm
+
+namespace modm::platform
 {
-	namespace platform
+	/**
+	 * @brief		Comparator Class for STM32
+	 *
+	 * @internal
+	 * @ingroup		{{partname}}_comp
+	 */
+	class Comp{{ id }} : public CompBase
 	{
-		/**
-		 * @brief		Comparator Class for STM32F3 series
-		 *
-		 * The class provides an interface to the comparators 1-7 availably on
-		 * ST's STM32F3 microcontroller series.
-		 *
-		 * @internal
-		 * @ingroup		{{partname}}_comp
-		 */
-		class Comp{{ id }}
+	public:
+		enum class
+		InvertingInput
 		{
-		public:
-			enum Mode
-			{
-				ULTRA_LOW_POWER = 0b00,
-				LOW_POWER = COMP{{ id }}_CSR_COMP{{ id }}MODE_0,		//0b01
-				MEDIUM_POWER = COMP{{ id }}_CSR_COMP{{ id }}MODE_1,		//0b10
-				HIGH_SPEED = 	COMP{{ id }}_CSR_COMP{{ id }}MODE_0 |
-								COMP{{ id }}_CSR_COMP{{ id }}MODE_1		//0b11
-			};
-
-			enum InvertingInput
-			{
-				VREF_1_4 = 0b000,	//0b000 1/4 of Vrefint
-				VREF_1_2 =			// 1/2 of Vrefint
-							COMP{{ id }}_CSR_COMP{{ id }}INSEL_0,	//0b001
-				VREF_3_4 =			// 3/4 of Vrefint
-							COMP{{ id }}_CSR_COMP{{ id }}INSEL_1,	//0b010
-				VREF = 				// Vrefint
-							COMP{{ id }}_CSR_COMP{{ id }}INSEL_1 |
-							COMP{{ id }}_CSR_COMP{{ id }}INSEL_0,	//0b011
-				// Common Port Aliases
-				PA4 =	COMP{{ id }}_CSR_COMP{{ id }}INSEL_2,		//0b100
-				DAC1 =	COMP{{ id }}_CSR_COMP{{ id }}INSEL_2,		//0b100
-				PA5 = 	COMP{{ id }}_CSR_COMP{{ id }}INSEL_2 |
-						COMP{{ id }}_CSR_COMP{{ id }}INSEL_0,		//0b101
-				DAC2 = 	COMP{{ id }}_CSR_COMP{{ id }}INSEL_2 |
-						COMP{{ id }}_CSR_COMP{{ id }}INSEL_0,		//0b101
+			{% if target.family == "f3" -%}
+			Vref1Div4		= 0b000 << 4,
+			Vref1Div2		= 0b001 << 4,
+			Vref3Div4		= 0b010 << 4,
+			Vref			= 0b011 << 4,
+			Dac1Channel1	= 0b100 << 4,
+			GpioA4			= 0b100 << 4,
+			Dac1Channel2	= 0b101 << 4,
+			GpioA5			= 0b101 << 4,
```

Comment:
What's the relation between this information and the one encoded in the `gpio` driver in the device file?
For the [F303x{bcde}](https://github.com/modm-io/modm-devices/blob/develop/devices/stm32/stm32f3-03-b_c_d_e.xml#L266-L307), I find this data in the files:

```xml
      <gpio port="a" pin="4">
        <signal driver="adc" instance="2" name="in1"/>
        <signal driver="comp" instance="1" name="inm"/>
        <signal driver="comp" instance="2" name="inm"/>
        <signal driver="comp" instance="3" name="inm"/>
        <signal driver="comp" instance="4" name="inm"/>
        <signal driver="comp" instance="5" name="inm"/>
        <signal driver="comp" instance="6" name="inm"/>
        <signal driver="comp" instance="7" name="inm"/>
        <signal device-size="b|c" driver="dac" name="out1"/>
        <signal device-size="d|e" driver="dac" instance="1" name="out1"/>
        <signal driver="opamp" instance="4" name="vinp"/>
        <signal driver="opamp" instance="4" name="vinp_sec"/>
        <signal af="2" driver="tim" instance="3" name="ch2"/>
        <signal af="3" driver="tsc" name="g2_io1"/>
        <signal af="5" driver="spi" instance="1" name="nss"/>
        <signal af="6" driver="i2s" instance="3" name="ws"/>
        <signal af="6" driver="spi" instance="3" name="nss"/>
        <signal af="7" driver="usart" instance="2" name="ck"/>
      </gpio>
      <gpio port="a" pin="5">
        <signal driver="adc" instance="2" name="in2"/>
        <signal driver="comp" instance="1" name="inm"/>
        <signal driver="comp" instance="2" name="inm"/>
        <signal driver="comp" instance="3" name="inm"/>
        <signal driver="comp" instance="4" name="inm"/>
        <signal driver="comp" instance="5" name="inm"/>
        <signal driver="comp" instance="6" name="inm"/>
        <signal driver="comp" instance="7" name="inm"/>
        <signal device-size="b|c" driver="dac" name="out2"/>
        <signal device-size="d|e" driver="dac" instance="1" name="out2"/>
        <signal driver="opamp" instance="1" name="vinp"/>
        <signal driver="opamp" instance="1" name="vinp_sec"/>
        <signal driver="opamp" instance="2" name="vinm"/>
        <signal driver="opamp" instance="2" name="vinm_sec"/>
        <signal driver="opamp" instance="3" name="vinp"/>
        <signal driver="opamp" instance="3" name="vinp_sec"/>
        <signal af="1" driver="tim" instance="2" name="ch1"/>
        <signal af="1" driver="tim" instance="2" name="etr"/>
        <signal af="3" driver="tsc" name="g2_io2"/>
        <signal af="5" driver="spi" instance="1" name="sck"/>
      </gpio>
```

Reply:
The Gpio signal data from the device files only contains the information that the gpio pin can be used as an input or output of a comparator instance.
Information about how to set the `COMP*(E)SEL` bits in the `COMPx->CSR` register is additionally needed to configure the comparator.